spec/01-unit/03-conf_loader_spec.lua

```diff
@@ -2686,4 +2686,70 @@ describe("Configuration loader", function()
     end)
   end)
 
+  describe("pluginserver config", function()
+    describe("fails if", function()
+      it("no query_command and not found in default location", function()
+        local _, err = conf_loader(nil, {
+          pluginserver_names = "gopher",
+          -- query_command = {},
+        })
+        assert.is_not_nil(err)
+        assert.matches("query_command undefined for pluginserver gopher", err)
+      end)
+
+      -- TODO need one more test case:
+      --   if no query_command is given, `/usr/local/bin/<plugi name>` is checked
+      --   we cannot write there in tests - figure out a way to write the test
+    end)
+    describe("warns if", function()
+      it("no start_command (meaning process is externally maintained)", function()
+        local spy_log = spy.on(log, "warn")
+
+        finally(function()
+          log.warn:revert()
+          assert:unregister("matcher", "str_match")
+        end)
+
+        assert:register("matcher", "str_match", function (_state, arguments)
+          local expected = arguments[1]
+          return function(value)
+            return string.match(value, expected) ~= nil
+          end
+        end)
+
+        local _, err = conf_loader(nil, {
+          pluginserver_names = "gopher",
+          pluginserver_gopher_query_cmd = "gopher -dump",
+        })
+        assert.is_nil(err)
+        assert.spy(spy_log).was_called(1)
+        assert.spy(spy_log).was_called_with("start_command undefined for pluginserver gopher; assuming external process management")
+      end)
+    end)
+    it("fills in default settings", function()
+      local conf, err = conf_loader(nil, {
+        pluginserver_names = "gopher",
+        pluginserver_gopher_query_cmd = "gopher -dump",
+        pluginserver_gopher_start_cmd = "gopher -p $KONG_PREFIX",
```

Comment:
Should we NOT specify start_cmd and ensure it is defaulted in this test?

Reply:
good catch; this is where the TODO note belongs - the default for `query_command` and `start_command` only apply if the server for a given plugin is in the default path (/usr/local/bin). we cannot reliably write in there, so I added one constant to `kong.conf_loader.constats` and mocked it in tests to test the defaults